stock_prices/stock_prices.py

```diff
@@ -3,7 +3,40 @@
 import argparse
 
 def find_max_profit(prices):
-  pass
+  # 9 lines of code?
+  # Find the maximum difference between smallest and largest price in list
+  # current_min_price_so_far and the max_profit_so_far?
+  # min price has to come before --> make a loop
+  # assign first value to min price so far...
+  # - min price --> [0] in array
+  # max price (currently on) - min price
+  # max profit 
+  # loop: if you find price smaller than current price - set that to be smallest - else continue with that mic price
+  # curr profit just calculates  --> currr profit net i in the list minus the min - price
+  # if price at index we're at is lower than min price - set it to be the min price
+  
+  # max_price = max(prices[ :])
+  # min_price = min(prices[ :maxpriceindex])
+  # get index of max value
+  # max value of array, ignoring [0]
+  # min value of prices up to the index (min, slice up to index)
+  # --> 
+  # take two values away from each other
+
+  current_min_price = prices[0]
+  max_profit_so_far = prices[1] - prices[0]
+  for i in range(len(prices) - 1):
+    if current_min_price > prices[i]:
+      current_min_price = prices[i]
+    else: 
+      current_min_price = current_min_price
+    current_profit = prices[i + 1] - current_min_price
+    if current_profit > max_profit_so_far:
+      max_profit_so_far = current_profit
+    else:
+      max_profit_so_far = max_profit_so_far
+
+  return max_profit_so_far
```

Comment:
On running the code against the tests, I noticed one test didn't pass.
self.assertEqual(find_max_profit([100, 55, 4, 98, 10, 18, 90, 95, 43, 11, 47, 67, 89, 42, 49, 79]), 94)